pkg/disttask/framework/scheduler/scheduler.go

```diff
@@ -584,7 +598,7 @@ func (s *BaseScheduler) markSubTaskCanceledOrFailed(ctx context.Context, subtask
 		if ctx.Err() != nil && context.Cause(ctx) == ErrCancelSubtask {
 			logutil.Logger(s.logCtx).Warn("subtask canceled", zap.Error(err))
 			s.updateSubtaskStateAndError(subtask, proto.TaskStateCanceled, nil)
-		} else if common.IsRetryableError(err) {
+		} else if common.IsRetryableError(err) || isRetryableError(err) {
```

Comment:
`common.IsRetryableError` already handles `ErrPDServerTimeout`. Need to find the root cause

Reply:
I think you should enhance `common.IsRetryableError`, rather than add another checking layer.

Reply:
> I think you should enhance `common.IsRetryableError`, rather than add another checking layer.

I think we should implement IsRetryableError for each business instead of enhance IsRetryableError since this works for Lightning but not always works for other business.

Reply:
then you should delete `common.IsRetryableError`?

Reply:
> then you should delete `common.IsRetryableError`?

Yes